src/main/java/budgettrackerapp/service/expenditure/SimpleExpenditureService.java

```diff
@@ -0,0 +1,95 @@
+package budgettrackerapp.service.expenditure;
+
+import budgettrackerapp.dto.CategoryDTO;
+import budgettrackerapp.dto.ExpenditureDTO;
+import budgettrackerapp.entity.Category;
+import budgettrackerapp.entity.Expenditure;
+import budgettrackerapp.entity.User;
+import budgettrackerapp.exeptions.ExpenditureInSpecificCategoryDoesNotExistException;
+import budgettrackerapp.mapper.CategoryMapper;
+import budgettrackerapp.mapper.UserMapper;
+import budgettrackerapp.repository.expenditure.ExpenditureRepository;
+import budgettrackerapp.repository.user.UserRepository;
+import budgettrackerapp.service.category.CategoryService;
+import budgettrackerapp.service.user.UserService;
+import lombok.AllArgsConstructor;
+import budgettrackerapp.mapper.ExpenditureMapper;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.math.BigDecimal;
+import java.util.Objects;
+
+@Service
+@AllArgsConstructor
+@Transactional
+public class SimpleExpenditureService implements ExpenditureService {
+
+    private ExpenditureRepository expenditureRepository;
+    private CategoryService categoryService;
+    private UserRepository userRepository;
+    private ExpenditureMapper expenditureMapper;
+    private CategoryMapper categoryMapper;
+
+    @Override
+    public ExpenditureDTO createAndAddToCategory(ExpenditureDTO expenditureDto, Long categoryId, Long userId) {
+        Objects.requireNonNull(expenditureDto, "Please, provide name and amount for your expenditure!");
+        Objects.requireNonNull(expenditureDto, "Id must be provided for this operation!");
+        CategoryDTO categoryDto = categoryService.getById(categoryId, userId);
+        User user = userRepository.findById(categoryDto.getUserId()).get();
+        Expenditure expenditure = expenditureMapper.mapToEntity(expenditureDto);
+
+        BigDecimal amountOfExpenditure = expenditure.getAmount();
+        user.setBalance((user.getBalance()).subtract(amountOfExpenditure));
+        userRepository.save(user);
+
+        expenditure.setCategory(categoryMapper.mapToEntity(categoryDto));
+
+        categoryService.updateBalance(categoryDto.getAmount().add(amountOfExpenditure), categoryDto);
+
+        Expenditure savedExpenditure = expenditureRepository.save(expenditure);
+        return expenditureMapper.mapToDto(savedExpenditure);
+    }
+
+    @Override
+    public void delete(Long categoryId, Long expenditureId, Long userId) {
+        Objects.requireNonNull(categoryId, "Category id must be provided for this operation!");
+        Objects.requireNonNull(expenditureId, "Expenditure id must be provided for this operation!");
```

Comment:
Make the error messages clearer - provided values must not be null, so let the client know about that